Major update: tree role (#12979)
mdn/content: files/en-us/web/accessibility/aria/roles/tree_role/index.md

  - ARIA roles
  - Tree role
  - composite widget role
  - needs example
---

A `tree` is a widget that allows the user to select one or more items from a hierarchically organized collection.

## Description

A `tree` widget is a hierarchical list with parent and child nodes that can expand and collapse. Any item in the hierarchy may have child tree items, set with [`role="treeitem"`](/en-US/docs/Web/Accessibility/ARIA/Roles/treeitem_role). Tree items that have children can be expanded or collapsed, showing and hiding their children. 

An example of a `tree` is a file system selection user interface: a tree view displaying folders and files. Folder items can be expanded to reveal the contents of the folder -- which may be files, folders, or both -- and collapsed, hiding its contents. 

> ARIA tree views are navigated primarily with arrow keys on the keyboard instead of the <kbd>tab</kbd> key which is not common for browser content, though normal and expected keyboard in native applications. For this reason, before creating a tree view, consider alternative options to address the functionality you need. 

Reviewer: ```suggestion
ARIA tree views are navigated primarily with arrow keys on the keyboard instead of the <kbd>Tab</kbd>. This form of navigation is not common for most browser content, though normal and expected for native applications. For this reason, before creating a tree view, consider alternative options to address the functionality you need. 
```
